visidata/graph.py

```diff
@@ -12,35 +12,75 @@ def numericCols(vd, cols):
 class InvertedCanvas(Canvas):
     def zoomTo(self, bbox):
         super().zoomTo(bbox)
-        self.fixPoint(Point(self.plotviewBox.xmin, self.plotviewBox.ymax), bbox.xymin)
-
-    def plotpixel(self, x, y, attr, row=None):
-        y = self.plotviewBox.ymax-y
-        self.pixels[y][x][attr].append(row)
+        self.fixPoint(Point(self.plotviewBox.xmin, self.plotviewBox.ymax),
+                      Point(bbox.xmin, bbox.ymax + 1/4*self.canvasCharHeight))
 
     def scaleY(self, canvasY):
-        'returns plotter y coordinate, with y-axis inverted'
-        plotterY = super().scaleY(canvasY)
-        return (self.plotviewBox.ymax-plotterY+4)
+        'returns a plotter y coordinate for a canvas y coordinate, with the y direction inverted'
+        return self.plotviewBox.ymax-round((canvasY-self.visibleBox.ymin)*self.yScaler)
 
-    def canvasH(self, plotterY):
-        return (self.plotviewBox.ymax-plotterY)/self.yScaler
+    def unscaleY(self, plotterY_inverted):
+        'performs the inverse of scaleY, returns a canvas y coordinate'
+        return (self.plotviewBox.ymax-plotterY_inverted)/self.yScaler + self.visibleBox.ymin
 
     @property
     def canvasMouse(self):
         p = super().canvasMouse
-        p.y = self.visibleBox.ymin + (self.plotviewBox.ymax-self.plotterMouse.y)/self.yScaler
+        p.y = self.unscaleY(self.plotterMouse.y)
         return p
 
+    def render_sync(self):
```

Comment:
I guess this is almost-but-not-exactly identical to Canvas.render_sync?  Is there any reasonable way to parameterize it for the inverted canvas, without duplicating the code?